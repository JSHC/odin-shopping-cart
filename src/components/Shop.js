import ShopItem from "./ShopItem";
import '../styles/Shop.css';

import image1 from '../images/green-t-shirt.jpg';
import image2 from '../images/cobalt-blue-t-shirt.jpg'
import image3 from '../images/purple-t-shirt.jpg';
import image4 from '../images/backpack-in-black.jpg';
import image5 from '../images/pair-of-navy-blue-skate-shoes.jpg'
import image6 from '../images/pair-of-black-and-white-sneakers.jpg'
import image7 from '../images/street-wear-navy-jacket.jpg'
import image8 from '../images/zip-up-jacket-for-men.jpg'

const Shop = (props) => {
    const shopItems = [
        {id: 1, name: 'Green T-Shirt', price: 10, image: image1},
        {id: 2, name: 'Blue T-Shirt', price: 10, image: image2},
        {id: 3, name: 'Purple T-Shirt', price: 10, image: image3},
        {id: 4, name: 'Black Backpack', price: 30, image: image4},
        {id: 5, name: 'Blue Shoes', price: 70, image: image5},
        {id: 6, name: 'Black and White Shoes', price: 75, image: image6},
        {id: 7, name: 'Navy Blue Jacket', price: 65, image: image7},
        {id: 8, name: 'Black Jacket', price: 70, image: image8},
    ]

    const onAddToCart = (item) => {
        if (props.onAddToCart) {
            props.onAddToCart(item);
        }
    }

    return (
        <div className="shop-container">
        <div className="shop-items-container">
            {shopItems.map(item => {
                return <ShopItem {...item} onAddToCart={onAddToCart} key={item.id}/>
            })}
        </div>
    </div>
    )
}

export default Shop;