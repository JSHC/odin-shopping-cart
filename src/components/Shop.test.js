import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Shop from "./Shop";

describe("Shop component", () => {
    it("renders", () => {
        const { container } = render(<Shop />);
        expect(container).toMatchSnapshot();
    });
});
