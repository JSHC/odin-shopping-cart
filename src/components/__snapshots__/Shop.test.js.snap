// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders 1`] = `
<div>
  <div
    class="shop-container"
  >
    <div
      class="shop-items-container"
    >
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(green-t-shirt.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Green T-Shirt
          </p>
          <p
            class="shop-item-price"
          >
            10
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(cobalt-blue-t-shirt.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Blue T-Shirt
          </p>
          <p
            class="shop-item-price"
          >
            10
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(purple-t-shirt.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Purple T-Shirt
          </p>
          <p
            class="shop-item-price"
          >
            10
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(backpack-in-black.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Black Backpack
          </p>
          <p
            class="shop-item-price"
          >
            30
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(pair-of-navy-blue-skate-shoes.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Blue Shoes
          </p>
          <p
            class="shop-item-price"
          >
            70
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(pair-of-black-and-white-sneakers.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Black and White Shoes
          </p>
          <p
            class="shop-item-price"
          >
            75
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(street-wear-navy-jacket.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Navy Blue Jacket
          </p>
          <p
            class="shop-item-price"
          >
            65
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        class="shop-item"
      >
        <div
          class="shop-item-image"
          style="background-image: url(zip-up-jacket-for-men.jpg);"
        />
        <div
          class="shop-item-details"
        >
          <p
            class="shop-item-name"
          >
            Black Jacket
          </p>
          <p
            class="shop-item-price"
          >
            70
          </p>
        </div>
        <div
          class="shop-item-controls"
        >
          <button
            aria-label="decrease-quantity"
          >
            -
          </button>
          <input
            aria-label="quantity"
            id="shop-item-quantity"
            name="shop-item-quantity"
            size="1"
            type="text"
            value="0"
          />
          <button
            aria-label="increase-quantity"
          >
            +
          </button>
          <button
            class="add-to-cart-button"
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
